mesonbuild/compilers/rust.py

```diff
@@ -153,6 +154,12 @@ def get_options(self) -> 'MutableKeyedOptionDictType':
             ),
         }
 
+    def get_dependency_compile_args(self, dep: 'Dependency') -> T.List[str]:
+        # Rust doesn't have dependency compile arguments so simply return
+        # nothing here. Dependencies are linked and all required metadata is
+        # provided by the linker flags.
+        return []
```

Comment:
rustc does not have any compiler args? This is more generic than just include paths, there are really no arguments that we could want to pass to rustc?

Reply:
To expand on this a bit, with C a dependency has to give compiler arguments to a dependent (`-I`, `-D`, or `-pthread` or whatever) and this is mostly because of the split between headers and library. You need to tell your dependent how to correctly feed your headers through the preprocessor.

With Rust the compiled crates (rlibs) are self-contained and contain all information needed to use them for further compilation.

That's what your question was about @xclaesse, or did I miss the point?

Reply:
Yes thanks, I just wanted to be sure we shouldn't instead filter some known flags. LGTM.